tests/Behat/Context/Api/WishlistContext.php

```diff
@@ -0,0 +1,354 @@
+
+        $response = $this->client->request(
+            self::DELETE,
+            sprintf('%s%s', self::$domain, $uri),
+            $this->getOptions(self::DELETE,[])
+        );
+
+        Assert::eq($response->getStatusCode(), 204);
+    }
+
+    /** @Then user tries to add product :product to the wishlist */
+    public function userTriesToAddProductToTheWishlist(ProductInterface $product)
+    {
+        $response = $this->addProductToTheWishlist($this->wishlist, $product);
+
+        $statusCode = $response->getStatusCode();
+
+        $this->resolveStatusCodeForUnauthenticatedUser($this->user, $statusCode);
+    }
+
+    /** @Then user tries to add :variant product variant to the wishlist */
+    public function userTriesToAddProductVariantToTheWishlist(ProductVariantInterface $variant)
+    {
+        $response = $this->addProductVariantToTheWishlist($this->wishlist, $variant);
+        $statusCode = $response->getStatusCode();
+
+        $this->resolveStatusCodeForUnauthenticatedUser($this->user, $statusCode);
+    }
+
+    /** @Then user removes :variant product variant from the wishlist
+     * @throws GuzzleException
+     */
+    public function userRemovesProductVariantFromTheWishlist(ProductVariantInterface $variant)
+    {
+        $uri = $this->router->generate('api_wishlists_shop_remove_product_variant_from_wishlist_item', [
+            "token" => $this->wishlist->getToken(),
+            "productVariantId" => $variant->getId()
+        ]);
+
+        $response = $this->client->request(
+            self::DELETE,
+            sprintf('%s%s', self::$domain, $uri),
+            $this->getOptions(self::DELETE)
+        );
+
+        Assert::eq($response->getStatusCode(), 204);
+    }
+
+    /** @Then user tries to remove product :product from the wishlist */
+    public function userTriesToRemoveProductFromTheWishlist(ProductInterface $product)
+    {
+        $response = $this->removeProductFromTheWishlist($this->wishlist, $product);
+        $statusCode = $response->getStatusCode();
+
+        $this->resolveStatusCodeForUnauthenticatedUser($this->user, $statusCode);
+    }
+
+    /** @Then user should have an empty wishlist */
+    public function userShouldHaveAnEmptyWishlist()
+    {
+        /** @var WishlistInterface $wishlist */
+
+        if (isset($this->user)) {
+            $wishlist = $this->wishlistRepository->findByShopUser($this->user);
+            var_dump(sprintf("Got user id : %d, wishlist id : %d",$this->user->getId(),$wishlist->getId()));
+        } else {
+            $wishlist = $this->wishlistRepository->find($this->wishlist->getId());
+            var_dump(sprintf("Empty user, wishlist id : %d",$wishlist->getId()));
+        }
+
+        Assert::eq(count($wishlist->getProducts()), 0);
+    }
+
+    /**
+     * @BeforeScenario
+     */
+    public function setupDomain()
+    {
+        $domain = (string)$this->getMinkParameter("base_url");;
+        self::$domain = trim($domain, "/");
+    }
+}
```

Comment:
- return type is missing
- public methods should be ahead of private ones
- var_dump to remove 
- behat annotations have wrong format, see how it is on other plugins
- why you sometimes use apostdores and then quotes, stick to one format
- ECS is missing